apps/examples/net_local/net_local_main.c

```diff
@@ -0,0 +1,151 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <strings.h>
+#include <unistd.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sched.h>
+#include <ctype.h>
+#include <pthread.h>
+
+#define MAXLINE 80
+
+extern int net_local_server_main(int argc, char *argv[]);
+extern int net_local_client_main(int argc, char *argv[]);
+
+int net_local_server_start(void)
+{
+	int pid = task_create("net_local_server", SCHED_PRIORITY_DEFAULT, 4096, net_local_server_main, (FAR char *const *)NULL);
+	if (pid < 0) {
+		printf("net_local_server start failed, err = %d\n", pid);
+	}
+	return pid;
+}
+
+int net_local_client_start(void)
+{
+	int pid = task_create("net_local_client", SCHED_PRIORITY_DEFAULT, 4096, net_local_client_main, (FAR char *const *)NULL);
+	if (pid < 0) {
+		printf("net_local_client start failed, err = %d\n", pid);
+	}
+	return pid;
+}
+
+int client_talk(int sockfd, const char *func, const char *str)
+{
+	int n;
+	char buf[MAXLINE + 1];
+	printf("[%s] send: %s\n", func, str);
+	write(sockfd, (const void *)str, strlen(str));
+
+	n = read(sockfd, buf, MAXLINE);
+	if (n < 0) {
+		perror("read");
+		return -1;
+	} else {
+		buf[n] = '\0';
+		printf("[%s] recv: %s\n", func, buf);
+		return strncasecmp(buf, str, n);
+	}
+}
+
+void server_talk(int sockfd, const char *func)
```

Comment:
In my opinion, return should be considered to check it worked properly or not.(void -> int or bool)